### PR TITLE
Add proxy options to templates_import module

### PR DESCRIPTION
**Pull request by adamlazik1**
relates to https://github.com/theforeman/foreman_templates/pull/191.

### CODE REVIEW

File: plugins/modules/templates_import.py
Comment (evgeni) [inline plugins/modules/templates_import.py]:
```suggestion
  http_proxy:
    description:
      - HTTP proxy to use for template sync.
      - Use this parameter together with C(http_proxy_policy=selected)
    required: false
    type: str
```

File: plugins/modules/templates_import.py
Comment (evgeni) [inline plugins/modules/templates_import.py]:
```suggestion
            http_proxy=dict(type='entity'),
```

File: tests/test_playbooks/tasks/templates_import.yml
Comment (evgeni) [inline tests/test_playbooks/tasks/templates_import.yml]:
```suggestion
    http_proxy: "{{ templates_import_http_proxy | default(omit) }}"
```

**Review by evgeni — CHANGES_REQUESTED** [review]
You don't have to pass IDs around, our abstraction layer can resolve entities for you :)

I've proposed changes (untested) that should do exactly that, treat "http_proxy" as an entity, and look it up via the API if passed.

**Comment by evgeni** [comment]
the diff looks good to me!
thanks!

**Comment by adamlazik1** [comment]
Hey @evgeni , I tried the patch but it does not seem to work:
```
ansible-playbook -i ,localhost -c local -e foreman_username=admin -e foreman_password=changeme -e templates_import_http_proxy='https proxy' -e templates_import_http_proxy_policy=selected -e foreman_server_url=http://foreman-dev.example.com -e foreman_validate_certs=false test.yaml  --module-path='foreman-ansible-modules/plugins/modules'

PLAY [all] ***********************************************************************************************************************************************************************************************************************************

TASK [Gathering Facts] ***********************************************************************************************************************************************************************************************************************
ok: [localhost]

TASK [Sync templates in from a git repo] *****************************************************************************************************************************************************************************************************
[WARNING]: You have configured a plain HTTP server URL. All communication will happen unencrypted.
fatal: [localhost]: FAILED! => {"changed": false, "error": {"message": "Couldn't find HttpProxy without an ID"}, "msg": "Error while performing import on templates: 422 Client Error: Unprocessable Content for url: http://foreman-dev.example.com/api/templates/import"}

PLAY RECAP ***********************************************************************************************************************************************************************************************************************************
localhost                  : ok=1    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0   
```
The http proxy id param did not get passed to the API. Any idea what could be the issue?

**Comment by adamlazik1** [comment]
> Hey @evgeni , I tried the patch but it does not seem to work:
> 
> ```
> ansible-playbook -i ,localhost -c local -e foreman_username=admin -e foreman_password=changeme -e templates_import_http_proxy='https proxy' -e templates_import_http_proxy_policy=selected -e foreman_server_url=http://foreman-dev.example.com -e foreman_validate_certs=false test.yaml  --module-path='foreman-ansible-modules/plugins/modules'
> 
> PLAY [all] ***********************************************************************************************************************************************************************************************************************************
> 
> TASK [Gathering Facts] ***********************************************************************************************************************************************************************************************************************
> ok: [localhost]
> 
> TASK [Sync templates in from a git repo] *****************************************************************************************************************************************************************************************************
> [WARNING]: You have configured a plain HTTP server URL. All communication will happen unencrypted.
> fatal: [localhost]: FAILED! => {"changed": false, "error": {"message": "Couldn't find HttpProxy without an ID"}, "msg": "Error while performing import on templates: 422 Client Error: Unprocessable Content for url: http://foreman-dev.example.com/api/templates/import"}
> 
> PLAY RECAP ***********************************************************************************************************************************************************************************************************************************
> localhost                  : ok=1    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0   
> ```
> 
> The http proxy id param did not get passed to the API. Any idea what could be the issue?

:facepalm: nevermind I forgot to update my test file. Yeah it works as intended so now I just wait for linked PR to get merged before I mark this ready for merge.

```
ansible-playbook -i ,localhost -c local -e foreman_username=admin -e foreman_password=changeme -e 'templates_import_http_proxy="https proxy"' -e templates_import_http_proxy_policy=selected -e foreman_server_url=http://foreman-dev.example.com -e foreman_validate_certs=false test.yaml  --module-path='foreman-ansible-modules/plugins/modules'

PLAY [all] ***********************************************************************************************************************************************************************************************************************************

TASK [Gathering Facts] ***********************************************************************************************************************************************************************************************************************
ok: [localhost]

TASK [Sync templates in from a git repo] *****************************************************************************************************************************************************************************************************
[WARNING]: You have configured a plain HTTP server URL. All communication will happen unencrypted.
changed: [localhost]

PLAY RECAP ***********************************************************************************************************************************************************************************************************************************
localhost                  : ok=2    changed=1    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
```

File: plugins/modules/templates_import.py
Comment (evgeni) [inline plugins/modules/templates_import.py:180]: You can add
```suggestion
        required_plugins=[('templates', ['*'])],
        required_if=[('http_proxy_policy', 'selected', ('http_proxy'))],
```
to enforce the requirement at ansible level too

**Comment by evgeni** [comment]
> 🤦 nevermind I forgot to update my test file. Yeah it works as intended so now I just wait for linked PR to get merged before I mark this ready for merge.

Phew. Your PR to templates looks correct, and the same lookup works in many other places, so it would be very weird if it wouldn't work here.

File: plugins/modules/templates_import.py
Comment (evgeni) [inline plugins/modules/templates_import.py]:
```suggestion
        required_if=[('http_proxy_policy', 'selected', ['http_proxy'])],
```

I'm stupid, sorry.

Comment (adamlazik1) [reply plugins/modules/templates_import.py]: No problem, I just figured it out myself too :smile:

**Review by evgeni — APPROVED** [review]

**Comment by adamlazik1** [comment]
This is now ready to be merged.

File: plugins/modules/templates_import.py
Comment (evgeni) [inline plugins/modules/templates_import.py]:
```suggestion
      - If you choose 'selected', provide the C(http_proxy) parameter.
```

File: plugins/modules/templates_import.py
Comment (evgeni) [inline plugins/modules/templates_import.py]:
```suggestion
      - You can choose no HTTP proxy, global HTTP proxy, or a custom HTTP proxy (C(selected))
```

**Review by evgeni — COMMENTED** [review]
minor doc fix, I'll apply this myself

**Comment by evgeni** [comment]
Whatever the codecov comment does here.

**Pull request merged**
